esx_service/vmdk_ops.py

```diff
@@ -634,9 +645,10 @@ def disk_detach(vmdk_path, vm):
     dev_changes.append(disk_spec)
     spec.deviceChange = dev_changes
 
+    # Handle the most generic exception for a VM reconfig.
     try:
         wait_for_tasks(si, [vm.ReconfigVM_Task(spec=spec)])
-    except vim.fault.GenericVmConfigFault as ex:
+    except vim.fault.VimFault as ex:
```

Comment:
Lets take a step back, once we catch the exception and prevent plugin crash whats going to happen is that the plugin has rightfully failed the request. Following can happen,

1. The VM (docker) may or may not have lost the context about the volume. If docker still remembers that the volume is attached then its a docker issue to detach the volume (doubt it though).

2. If docker doesn't have any context of the volume detach having failed, which would be a docker bug btw, then we can't expect the plugin to keep state of the failure. The volume remains attached (meta-data also) and will need something like VC/vim-cmd as the user chooses, to remove the volume from the Vm config.

3. The plugin is by design stateless w.r.t errors that happen beyond its control. Its like a VC user initiates a reconfig and the ESX crashes, we have no idea if the op went thru or not.

For this issue I see no more change except to handle the correct exception and avoid crashing.